### PR TITLE
add able to configure DB connection limits in sql backend

### PR DESCRIPTION
**Pull request by lord-alfred**
PR for https://github.com/flashmob/go-guerrilla/issues/174

I think you can make a description on the wiki:

**sql_max_open_conns** - sets the maximum number of open connections to the database. The default is 0 (unlimited).

**sql_max_idle_conns** - sets the maximum number of connections in the idle connection pool. The default is 2.

**sql_max_conn_lifetime** - sets the maximum amount of time a connection may be reused.

The sum of `sql_max_open_conns` and `sql_max_idle_conns` should not be greater than `max_connections` for MySQL or PostgreSQL. `sql_max_idle_conns` must be less than `sql_max_open_conns`.

`sql_max_conn_lifetime` must be less than `wait_timeout` in MySQL.

### CODE REVIEW

**Comment by lord-alfred** [comment]
@flashmob why Travis build failed? (I newbie in golang and dont understand where I broke syntax)

**Comment by flashmob** [comment]
It looks like you didn't run "go fmt" before committing.

https://blog.golang.org/go-fmt-your-code

😀

**Comment by lord-alfred** [comment]
Sorry, I didn’t know about it 😬

File: backends/p_sql.go [hunk 1]
```diff
@@ -40,12 +47,15 @@ func init() {
...
-	PrimaryHost string `json:"primary_mail_host"`
+	Table              string `json:"mail_table"`
+	Driver             string `json:"sql_driver"`
+	DSN                string `json:"sql_dsn"`
+	SQLInsert          string `json:"sql_insert,omitempty"`
+	SQLValues          string `json:"sql_values,omitempty"`
+	PrimaryHost        string `json:"primary_mail_host"`
+	MaxOpenConns       int    `json:"sql_max_open_conns,omitempty"`
+	MaxIdleConns       int    `json:"sql_max_idle_conns,omitempty"`
+	MaxConnLifetimeSec int    `json:"sql_max_conn_lifetime_sec,omitempty"`
```
Comment (flashmob) [inline backends/p_sql.go]: rename to `MaxConnLifetime` & use `string`

File: backends/p_sql.go [hunk 2]
```diff
@@ -60,6 +70,17 @@ func (s *SQLProcessor) connect() (*sql.DB, error) {
...
 		return nil, err
 	}
+
+	if s.config.MaxOpenConns != 0 {
+		db.SetMaxOpenConns(s.config.MaxOpenConns)
+	}
+	if s.config.MaxIdleConns != 0 {
+		db.SetMaxIdleConns(s.config.MaxIdleConns)
+	}
+	if s.config.MaxConnLifetimeSec != 0 {
```
Comment (flashmob) [inline backends/p_sql.go]: For MaxConnLifetimeSec, (after renamed to MaxConnLifetime) let's use `time.ParseDuration` to parse the string.
So then specifying time becomes more flexible
https://golang.org/pkg/time/#ParseDuration

**Comment by lord-alfred** [comment]
Improved what you wrote in the review and corrected the first post with info for wiki.

File: backends/p_sql.go (hunk 1 above)
Comment (lord-alfred) [reply backends/p_sql.go]: Done :)

File: backends/p_sql.go (hunk 2 above)
Comment (lord-alfred) [reply backends/p_sql.go]: Good, thanks :)

**Pull request merged**